vhost-device-sound/src/device.rs

```diff
@@ -984,8 +934,13 @@ mod tests {
             };
             let addr_req = 0x10_0000;
             let descs = [
-                Descriptor::new(addr_req, 0x100, 0, 0), // request
-                Descriptor::new(0x20_0000, 0x100, VRING_DESC_F_WRITE as u16, 0), // response
+                Descriptor::new(addr_req, size_of::<VirtioSoundHeader>() as u32, 0, 0), // request
+                Descriptor::new(
+                    0x20_0000,
+                    size_of::<VirtioSoundHeader>() as u32,
+                    VRING_DESC_F_WRITE as u16,
+                    0,
+                ), // response
```

Comment:
This probably could go into a separate commit? It does not seem to be related to the rest of the changes that happen here, or is there a relation?

Reply:
There is a relation. The test is checking for failure when a malformed control request has arrived, i.e., `ControlMessageKind::PcmInfo`, `ControlMessageKind::ChmapInfo`, `ControlMessageKind::JackInfo`. In the previous code, this required three descriptors so the test was creating two descriptors to check if the processing of the requests effectively fail. With the new code, the distribution over descriptors does not matter and the test was failing. To fix this, I had to reduce the len of the second descriptor to a value less than the expected payload for control msg above. Fixing this is somehow related with the commit. I do not know if I fixed it in the right way.

Reply:
IMHO it's fine to do here since it's related.